### ISSUE TITLE
`bcf` is not compatible with `musttail`

### ISSUE THREAD

**Issue by yujincheng08** [61bcdefg/Hikari-LLVM15#57]
```
  musttail call must precede a ret with an optional bitcast
    %7 = musttail call i32 @dlclose(ptr noundef %5) #33, !dbg !25865
  LLVM ERROR: Broken module found, compilation aborted!
```

**Comment by 61bcdefg** [comment]
Please provide the source code or the non-obfuscated IR

**Comment by yujincheng08** [comment]
@61bcdefg 

Example codes:

```c++
#include <dlfcn.h>

int my_dlclose(void *handle) {
    if (handle != nullptr) {
        [[clang::musttail]] return dlclose(handle);
    } else {
        return 1;
    }
}
```

**Comment by 61bcdefg** [comment]
https://github.com/61bcdefg/Hikari-LLVM15-Core/commit/1312b9fe4800ba3c89d5964b6dafcb8a686bb7cd

**Comment by yujincheng08** [comment]
let me try try

**Comment by yujincheng08** [comment]
```
/llvm/include/llvm/Transforms/Obfuscation/CryptoUtils.h:38:4: warning: extra ';' [-Wpedantic]
```

**Comment by 61bcdefg** [comment]
> ```
> 
> /llvm/include/llvm/Transforms/Obfuscation/CryptoUtils.h:38:4: warning: extra ';' [-Wpedantic]
> 
> ```

This is a minor syntax error that has no effect other than this warning

**Comment by yujincheng08** [comment]
I know. But just a kindly remind.

**Comment by yujincheng08** [comment]
It works now.
